### ISSUE TITLE
no-invalid-this fails when used with ES7 class property declaration

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#3475]
```js
class {
    toggleLocationActiveState = () => {
        this.props.onActivateLocation(this.props.id);
    }
}
```

```
curiosity:foo gajus$ eslint -v
v1.2.1
```

```
/foo/index.js
  13:9   error  Unexpected `this`                       no-invalid-this
  13:39  error  Unexpected `this`                       no-invalid-this
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by lo1tuma** [comment]
ESLint doesn’t support this experimental ECMAScript feature. I guess you are using `babel-eslint`, please file an issue there.

**Comment by julienw** [comment]
Now that this is at stage 3, do you think this could be added ?

**Comment by ilyavolodin** [comment]
ESLint only works with stage 4 features. For everything else you should use babel-eslint parser.

**Comment by julienw** [comment]
More accurately https://github.com/babel/eslint-plugin-babel :) Thanks for your answer
